src/gc/pooltable.d

```diff
@@ -0,0 +1,287 @@
+/**
+ * A sorted array to quickly lookup pools.
+ *
+ * Copyright: Copyright Digital Mars 2001 -.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Walter Bright, David Friedman, Sean Kelly, Martin Nowak
+ */
+module gc.pooltable;
+
+static import cstdlib=core.stdc.stdlib;
+
+struct PoolTable(Pool)
+{
+    import core.stdc.string : memmove;
+
+nothrow:
+    void reset()
```

Comment:
Maybe call this `Dtor` for consistency with other structs in the GC implementation?